src/ai/backend/test/testcases/session/creation_failure_low_resources.py

```diff
@@ -0,0 +1,43 @@
+from ai.backend.client.exceptions import BackendAPIError
+from ai.backend.client.output.fields import image_fields
+from ai.backend.common.types import ClusterMode
+from ai.backend.test.contexts.client_session import ClientSessionContext
+from ai.backend.test.contexts.image import ImageContext
+from ai.backend.test.templates.template import TestCode
+
+
+class SessionCreationFailureLowResources(TestCode):
+    async def test(self) -> None:
+        client_session = ClientSessionContext.current()
+        image = ImageContext.current()
+        session_name = "test-session-creation-failure"
+
+        result = await client_session.Image.get(
+            image.name, image.architecture, fields=[image_fields["labels"]]
+        )
+        labels = result["labels"]
+        min_mem_label = next(
+            filter(lambda label: label["key"] == "ai.backend.resource.min.mem", labels), None
+        )
+
+        if not min_mem_label:
+            raise RuntimeError(
+                "Minimum memory label not found in image labels. Please check the image configuration."
+            )
+
+        min_mem = min_mem_label["value"]
+
+        try:
+            await client_session.ComputeSession.get_or_create(
+                image.name,
+                architecture=image.architecture,
+                name=session_name,
+                # This failed due to the need for additional space for shared memory.
+                resources={"cpu": 1, "mem": min_mem},
+                cluster_mode=ClusterMode.SINGLE_NODE,
+                cluster_size=1,
+            )
+            assert False, "Expected BackendAPIError for low resources was not raised"
```

Comment:
raise `UnexpectedSuccess`.
Please use assert for value verification, and ensure that test failure cases other than value verification do not raise assertion errors.